### PR TITLE
Variable Timout for Katello Module + Documentation

### PR DESCRIPTION
**Pull request by FloThinksPi**
##### SUMMARY

This Change firstly introduces a Timeout variable to the katello module. 
This is needed als in many environments, the hardcoded 1000 Seconds are not sufficient resulting in an always failing Task.

Secondly Extended the Documentation to reflect all possible entity+actions+params combinations.

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
/modules/remote_management/foreman/katello.py

##### ANSIBLE VERSION

```
ansible 2.5.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/WN00074879/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]

```

##### ADDITIONAL INFORMATION
None

### CODE REVIEW

**Comment by FloThinksPi** [comment]
uhm think i need to look over the ci results first :)

File: .ansible-test/inventory.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+localhost ec2_kernel=aki-fc8f11cc ec2_state=running ec2_tag_Cname=testbox ec2_ami_launch_index=0 ec2_private_dns_name=ip-testbox.us-west-2.compute.internal ec2_instance_type=m3.medium ec2_sourceDestCheck=true ec2_id=i-b9999999 ec2_private_ip_address=localhost ec2_security_group_names=Web,default,SSH ec2_vpc_id=vpc-c9999999 ec2_state_code=16 ec2_tag_Name=testbox ec2_architecture=x86_64 ec2_hypervisor=xen ec2_key_name=mgkey ec2_root_device_type=ebs ec2_client_token=141031249511895261 ec2_public_dns_name=ec2-testbo.us-west-2.compute.amazonaws.com ec2_monitoring_state=disabled ec2_root_device_name=/dev/sda ec2_tag_Role=testrole ec2_subnet_id=subnet-b9999999 ec2_security_group_ids=sg-9bc871fe,sg-92d56cf7,sg-14cb7271 ec2_placement=us-west-2b ec2_launch_time=2014-09-10T01:28:20.000Z ec2_ip_address=127.0.0.1 ec2_image_id=ami-b9999999 ec2_region=us-west-2 ec2_tag_Level=dev ec2_virtualization_type=paravirtual ec2_tag_ShardKey=testing-001 ec2_dns_name=ec2-testbox.compute.amazonaws.com
```
Comment (mattclay) [inline .ansible-test/inventory.yml]: The `.ansible-test` directory and files appear to be unrelated to the PR and should be removed.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM debian:7.7
```
Comment (mattclay) [inline Dockerfile]: This file appears to be unrelated to the PR and should be removed.

**Comment by FloThinksPi** [comment]
@mattclay Tanks for the hint. I had trouble getting ansible-test to work locally and thus accidently commited a file from my tries to get ansible-test running. I removed it now and also got ansible-test working, so i was able to fix all CI findigs except the E309 finding. This is because the feature verify_ssl was already implemented but not documented. As well as:

```
E328 Choices value from the documentation ('name') is not compatible with type 'dict' defined in the argument_spec
```

but i dont know how do document the module better and use choices that are compatible with dicts.

**Review by ehelms — APPROVED** [review]
@FloThinksPi I do want to bring your attention to this PR I opened today -- https://github.com/ansible/ansible/pull/42043

**Comment by FloThinksPi** [comment]
@ehelms thanks for the hint, didnt know this module is going to replace the "official" community supported one :smiley: .

Concerning these Changes, i think this is more like a bugfix than a feature enrichment and maybe it makes sense to include it even in the depricated module ?

What do you think ? Include these Changes and also depricate the module with #42043 ?

**Review by ehelms — APPROVED** [review]
@FloThinksPi Agreed! I'm fine with any changes to help make these modules better in the interim. That's on me that we've not done better at broadcasting the newer module set and taking this long to deprecate these modules I originally wrote. I do think you'll find the newer modules more robust, easier to focus on entities, better testing overall and more community support behind them.

**Comment by FloThinksPi** [comment]
@mattclay can you review this PR once more ? 

The tag needs_revision is misleading here as we wont be able to fix all CI Issues in this old module without a whole rewrite, which was already done by the foreman community.

**Comment by mattclay** [comment]
@FloThinksPi Which CI issues are you referring to?

**Comment by FloThinksPi** [comment]
@mattclay The ones which where reported by anibot last time:
```
lib/ansible/modules/remote_management/foreman/katello.py:0:0: E309 version_added for new option (action) should be 2.7. Currently 0.0
lib/ansible/modules/remote_management/foreman/katello.py:0:0: E309 version_added for new option (verify_ssl) should be 2.7. Currently 0.0
lib/ansible/modules/remote_management/foreman/katello.py:0:0: E328 Choices value from the documentation ('name') is not compatible with type 'dict' defined in the argument_spec
test/sanity/validate-modules/ignore.txt:1089:1: A102 Remove since "lib/ansible/modules/remote_management/foreman/katello.py" passes "E322" test
test/sanity/validate-modules/ignore.txt:1090:1: A102 Remove since "lib/ansible/modules/remote_management/foreman/katello.py" passes "E325" test 
```

So in my opinion these are fine as the structure of the module does not allow to document it correctly with the choices method, as "params" is a dict, so ive written all choices in the comment of that parameter. Also the first 2 Issues are Ok because verify_ssl was and action where undocumented :) .

**Comment by mattclay** [comment]
@FloThinksPi The version_added errors can be ignored since those options are not new. The remaining issues need to be fixed:

1. Remove the choices from params (the updated description can remain) since params is a dict.
2. Remove references to `lib/ansible/modules/remote_management/foreman/katello.py` from the file `test/sanity/validate-modules/ignore.txt` since those errors have been corrected and no longer need to be ignored.

**Comment by FloThinksPi** [comment]
@mattclay all done
